### DISCUSSION TITLE
Support for post-process complex algorithm

### DISCUSSION THREAD

**Discussion by vincentsarago** [developmentseed/titiler#493] [Feature Request]
Following https://github.com/developmentseed/titiler/discussions/487, 

I think titiler is missing a key feature which is the support for complex `post-process` operation (e.g hillshading). Right now we can use `expression` to apply simple arithmetic operation but it's really limited to what numexpr can handle (https://numexpr.readthedocs.io/projects/NumExpr3/en/latest/user_guide.html#supported-operators).

In rio-viz (a titiler local like plugin) I've started a PR that sketches what it could be: https://github.com/developmentseed/rio-viz/pull/51

**Comment by vincentsarago** [comment]
on going work over https://github.com/developmentseed/titiler/pull/542

**Comment by jo-chemla** [comment]
For anyone stumbling on this discussion, this was merged into titiler via the use of algorithms (hillshade, contour or custom, see [this doc](https://github.com/developmentseed/titiler/blob/main/docs/src/user_guide/algorithms.md)

EDIT: frontend support in `/cog/viewer` to algorithms and expressions added in PR https://github.com/developmentseed/titiler/pull/1119
